src/beyond_identity/api/utils/request.rs

```diff
@@ -0,0 +1,155 @@
+use http::Method;
+use reqwest_middleware::ClientWithMiddleware as Client;
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::{
+    beyond_identity::{api_token::get_beyond_identity_api_token, tenant::TenantConfig},
+    common::{config::Config, error::BiError},
+};
+
+pub async fn send_request<T, U>(
```

Comment:
This function could be rewritten as a middleware for reqwest_middleware. See the example [here](https://docs.rs/reqwest-middleware/latest/reqwest_middleware/trait.Middleware.html#example).